### PR TITLE
Package with pbr

### PR DESCRIPTION
**Pull request by t184256**
An attempt at #16: packaging with pbr.

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Axel Huebl <axel.huebl@plasma.ninja> <axel.huebl@web.de>
```
Comment (ax3l) [inline .mailmap:1]: do we really need a mailmap?

If so, my private mail is the first and my work mail is [at hzdr.de](https://www.hzdr.de/db/!ContMan.Visi.Card?pNid=no&pUser=5135)

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[metadata]
+name = openPMD_validator
```
Comment (ax3l) [inline setup.cfg]: as with `openPMD-viewer` we should try to use the same naming :)
https://pypi.python.org/pypi/openPMD-viewer/0.7.1

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    Environment :: Console
+    Intended Audience :: Science/Research
+    License :: OSI Approved :: ISC License (ISCL)
+    Operating System :: OS Independent
+    Programming Language :: Python :: 2.7
+    Programming Language :: Python :: 3
+    Topic :: Scientific/Engineering :: Physics
+
+[files]
+packages = openPMD_validator
```
Comment (ax3l) [inline setup.cfg]: see above

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+[entry_points]
+console_scripts =
+    openPMD_check_h5 = openPMD_validator.check_h5:main
```
Comment (ax3l) [inline setup.cfg]: see above

File: .mailmap (hunk 1 above)
Comment (t184256) [reply .mailmap:1]: Well, one either learn to commit with a single email per project or uses a mailmap =)

I (pbr, actually) took those from the commit history.

**Review by ax3l — COMMENTED** [review]
thank you for the PR! ✨  added a few comments inline

File: .mailmap (hunk 1 above)
Comment (ax3l) [reply .mailmap:1]: Dang, we were young and did not know better ;)

No jokes aside, why do we need the mailmap for packaging? We can just add the authors manually, no?

Comment (t184256) [reply .mailmap:1]: As pbr puts it: "Why keep an AUTHORS or a ChangeLog file when git already has all of the information you need?"

If you don't care about duplicates in the AUTHORS file, the mailmap can be dropped.

File: setup.cfg (hunk 3 above)
Comment (t184256) [reply setup.cfg]: so it should be `opmd_validator`?

File: .mailmap (hunk 1 above)
Comment (ax3l) [reply .mailmap:1]: I see, thx!

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    Environment :: Console
+    Intended Audience :: Science/Research
+    License :: OSI Approved :: ISC License (ISCL)
+    Operating System :: OS Independent
+    Programming Language :: Python :: 2.7
+    Programming Language :: Python :: 3
+    Topic :: Scientific/Engineering :: Physics
+
+[files]
+packages = opmd_validator
```
Comment (ax3l) [inline setup.cfg]: ideally, `openPMD-validator` or more pythonic `openPMD_validator`

Comment (ax3l) [reply setup.cfg]: I wonder if such a thing such as sub-packages like `openPMD.validator`, `openPMD.viewer`, `openPMD.API`, ... exist?

Comment (t184256) [reply setup.cfg]: I'm not an expert, but I think that packages are mostly dashed, while modules are nearly universally underscored. Please settle on something and I'll be happy to use them.

Comment (t184256) [reply setup.cfg]: "I wonder if such a thing such as sub-packages like openPMD.validator, openPMD.viewer, openPMD.API, ... exist?" — Submodules shipped in different packages? Yes, of course it's used (and abused). Bear in mind that adopting this across several projects is a large external-facing change.

Comment (ax3l) [reply setup.cfg]: you are right, dashed name is good and underscored module, just let us use the full name `openpmd_` as prefix please. This is the naming scheme we want to change the viewer to as well: https://github.com/openPMD/openPMD-viewer/issues/82

Comment (t184256) [reply setup.cfg]: Oh, OK. Hope it'd all get squashed on merge.

File: .travis.yml [hunk 6]
```diff
@@ -25,14 +25,14 @@ install:
 before_script:
   # static code analysis
   #   pyflakes: mainly warnings, unused code, etc.
-  - python -m pyflakes .
+  - python -m pyflakes opmd_validator
```
Comment (ax3l) [inline .travis.yml]: I think these need an update as well for the package name

Comment (t184256) [reply .travis.yml]: Argh, it slipped through. Fixed.

**Comment by t184256** [comment]
I see that I broke travis builds, but I have no idea what I did wrong or how to fix that. Can you have a look?

**Comment by RemiLehe** [comment]
Thanks a lot for the pull request! 
I think I prefer this one to the vanilla one (#23)
I'll have a few suggestions, which I think should fix the Travis tests. I thought the easiest would be to do a PR to your branch: https://github.com/t184256/openPMD-validator/pull/1
If you accept and merge https://github.com/t184256/openPMD-validator/pull/1, it will automatically update the present page.

**Review by RemiLehe — APPROVED** [review]

**Comment by ax3l** [comment]
@RemiLehe shall I publish a package on PyPI? I could take the release management of this repo if you like.

**Comment by ax3l** [comment]
@RemiLehe I actually wanted to get your feedback (reviewer) to test the final version before merge (that's why I self assigned me ;) )

**Comment by RemiLehe** [comment]
Oops, sorry for merging early. I now see from your other commit that `pbr` is not ideally suited. Do you want me to do a PR to switch a more standard `setup.py` ?

**Comment by ax3l** [comment]
Saw it already. That's great, thank you! :)

**Pull request merged**